### PR TITLE
Update generated client design and clean up code

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject 

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/4231

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: examples/medical-center/generated/persist_client.bal [hunk 1]
```diff
@@ -3,22 +3,22 @@
...
+public client class Client {
     *persist:AbstractPersistClient;
 
     private final mysql:Client dbClient;
 
     private final map<persist:SQLClient> persistClients;
 
     private final record {|persist:Metadata...;|} metadata = {
-        "medicalitem": {
+        "medicalitems": {
```
Comment (daneshk) [inline examples/medical-center/generated/persist_client.bal:21]:
```suggestion
        [MEDICAL_ITEM]: {
```

File: examples/medical-center/generated/persist_client.bal [hunk 2]
```diff
@@ -29,7 +29,7 @@ public client class MedicalCenterClient {
             },
             keyFields: ["itemId"]
         },
-        "medicalneed": {
+        "medicalneeds": {
```
Comment (daneshk) [inline examples/medical-center/generated/persist_client.bal:32]:
```suggestion
        [MEDICAL_NEED]: {
```

File: examples/medical-center/generated/persist_client.bal [hunk 3]
```diff
@@ -50,62 +50,61 @@ public client class MedicalCenterClient {
         }
         self.dbClient = dbClient;
         self.persistClients = {
-            medicalitem: check new (self.dbClient, self.metadata.get(MEDICAL_ITEM)),
-            medicalneed: check new (self.dbClient, self.metadata.get(MEDICAL_NEED))
+            medicalitems: check new (self.dbClient, self.metadata.get(MEDICAL_ITEM)),
+            medicalneeds: check new (self.dbClient, self.metadata.get(MEDICAL_NEED))
```
Comment (daneshk) [inline examples/medical-center/generated/persist_client.bal:54]:
```suggestion
            [MEDICAL_ITEM]: check new (self.dbClient, self.metadata.get(MEDICAL_ITEM)),
            [MEDICAL_NEED]: check new (self.dbClient, self.metadata.get(MEDICAL_NEED))
```

**Review by daneshk — DISMISSED** [review]
LGTM

File: examples/medical-center/generated/persist_client.bal (hunk 3 above)
Comment (kaneeldias) [reply examples/medical-center/generated/persist_client.bal]: Fixed in [d0b65f0](https://github.com/ballerina-platform/module-ballerina-persist/pull/177/commits/d0b65f00336f1917043ae3b82e35b0c405656d9f)

File: examples/medical-center/generated/persist_client.bal (hunk 2 above)
Comment (kaneeldias) [reply examples/medical-center/generated/persist_client.bal]: Fixed in [d0b65f0](https://github.com/ballerina-platform/module-ballerina-persist/pull/177/commits/d0b65f00336f1917043ae3b82e35b0c405656d9f)

File: examples/medical-center/generated/persist_client.bal (hunk 1 above)
Comment (kaneeldias) [reply examples/medical-center/generated/persist_client.bal]: Fixed in [d0b65f0](https://github.com/ballerina-platform/module-ballerina-persist/pull/177/commits/d0b65f00336f1917043ae3b82e35b0c405656d9f)

**Review by daneshk — APPROVED** [review]

**Pull request merged**